src/main/java/cl/transbank/patpass/PatpassByWebpay.java

```diff
@@ -0,0 +1,145 @@
+
+        return PatpassByWebpay.getOptions().buildOptions(options);
+    }
+
+    public static class Transaction {
+        public static CreatePatpassByWebpayTransactionResponse create(
+                String buyOrder, String sessionId, double amount, String returnUrl, String serviceId, String cardHolderId,
+                String cardHolderName, String cardHolderLastName1, String cardHolderLastName2, String cardHolderMail, String cellphoneNumber,
+                String expirationDate, String commerceMail, boolean ufFlag) throws CreateTransactionException {
+            return Transaction.create(buyOrder, sessionId, amount, returnUrl, serviceId, cardHolderId, cardHolderName,
+                    cardHolderLastName1, cardHolderLastName2, cardHolderMail, cellphoneNumber, expirationDate, commerceMail, ufFlag, null);
+        }
+
+        public static CreatePatpassByWebpayTransactionResponse create(
+                String buyOrder, String sessionId, double amount, String returnUrl, String serviceId, String cardHolderId,
+                String cardHolderName, String cardHolderLastName1, String cardHolderLastName2, String cardHolderMail, String cellphoneNumber,
+                String expirationDate, String commerceMail, boolean ufFlag, Options options) throws CreateTransactionException {
+            try {
+                options = PatpassByWebpay.buildOptions(options);
+                final URL endpoint = new URL(getCurrentIntegrationTypeUrl(options.getIntegrationType()));
+                final TransactionCreateRequest request = new TransactionCreateRequest(buyOrder, sessionId, amount, returnUrl);
+                request.setDetails(serviceId, cardHolderId, cardHolderName, cardHolderLastName1, cardHolderLastName2,
+                        cardHolderMail, cellphoneNumber, expirationDate, commerceMail, ufFlag);
+                return WebpayApiResource.execute(endpoint, HttpUtil.RequestMethod.POST, request, options, CreatePatpassByWebpayTransactionResponse.class);
+            } catch (Exception e) {
+                throw new CreateTransactionException(e);
+            }
+        }
+
+        public static CommitPatpassByWebpayTransactionResponse commit(String token) throws CommitTransactionException {
+            return Transaction.commit(token, null);
+        }
+
+        public static CommitPatpassByWebpayTransactionResponse commit(String token, Options options) throws CommitTransactionException {
+            try {
+                options = PatpassByWebpay.buildOptions(options);
+                return BeanUtils.getInstance().copyBeanData(new CommitPatpassByWebpayTransactionResponse(), WebpayPlus.Transaction.commit(token, options));
+            } catch (Exception e) {
+                throw new CommitTransactionException(e);
+            }
+        }
+
+        public static RefundPatpassByWebpayTransactionResponse refund(String token, double amount) throws RefundTransactionException {
```

Comment:
We should ask about this before to merge this changes because the refund is not defined in Patpass by webpay

Reply:
I know, but I have alrwady did it and this merge is to the intermediate branch. I think it is better to merge this code there in case this will work in the API as now. When we merge this code to master I must create a new PR from intermediate branch to master and then we could clean code from this method, but if in the future we should implement them we could retrive them from GIT history. What you think? @a1b4

Reply:
sure sounds good 👍

Reply:
sure sounds good 👍